---
title: CV
layout: page
permalink: /cv/index.html
---
## Jacob C. Kimmel
[jacob@jck.bio](mailto:jacob@jck.bio)

San Francisco, California

[CV PDF]({{site.url}}/assets/cv.pdf)

## Education

**Ph.D. â€” Dept. Biochemistry & Biophysics, UC San Francisco, 2015 - 2018**  
*Funding*: NSF Fellowship, PhRMA Fellowship, NIH T32, UCSF Discovery    

<!-- **B.S. (Honors), Biotechnology, University of Central Florida, 2012 - 2015**  
*Distinctions*: Summa Cum Laude, Top Honors, University Honors  
*Funding*: National Merit Scholarship, Full Tuition & Stipend  
*GPA*: 4.0 / 4.0 -->

<!-- **Marine Biological Laboratory, Woods Hole, MA, 2016 Fall**  
*Course*: Computational Image Analysis in Cell and Developmental Biology -->

## Recent Experience

**NewLimit, South San Francisco, CA, 2022 - Present**  
*Co-founder & President*  

* Building a therapeutics firm to develop epigenetic reprogramming medicines for aging

**Calico Life Sciences, South San Francisco, CA, 2020 - 2022**  
*Principal Investigator, R&D, 2021-2022*  
*Computational Fellow, Computing, 2020-2021*

* Led a laboratory combining computational and experimental methods to repurpose developmental programs for therapeutic applications
* Developed [**scNym**](https://scnym.research.calicolabs.com) semi-supervised adversarial neural networks for classifying cell types in single cell genomics, improving the state-of-the-art -- [scnym.research.calicolabs.com](https://scnym.research.calicolabs.com)
* Led the development of a computational & experimental platform for pooled screening of cell identity reprogramming strategies, successfully completing a company-wide goal -- [reprog.research.calicolabs.com](https://reprog.research.calicolabs.com)

**Calico Life Sciences, South San Francisco, CA, 2018 - 2020**.  
*Data Scientist, Computing*  

* Led an [investigation of cell type and tissue environment influences on aging](http://jck.bio/murine_cell_aging) using single cell genomics across three murine tissues -- [mca.research.calicolabs.com](https://mca.research.calicolabs.com)
* Led an [investigation of skeletal muscle aging](https://myo.research.calicolabs.com) using single cell genomics & dynamical systems, revealing a decision point where old stem cells fail to differentiate  -- [myo.research.calicolabs.com](https://myo.research.calicolabs.com)
* Developed timelapse image analysis methods for oncology applications, enabling multi-cell tracking and analysis over many days
* Automated quantification of yeast cell aging using convolutional neural networks
* Built an automated targeting system for laser ablation microscopy with sub-millisecond timing

**University of California San Francisco, San Francisco, CA, 2015 - 2018**  
*PhD Candidate*  
*Principal Investigators:* Wallace Marshall, Andrew Brack  
*Thesis:* Inferring stem cell state from cell behavior

* Developed [*Heteromotility*](http://jck.bio/heteromotility) biological motion analysis package, including feature extraction, unsupervised clustering, and time-series analysis tools to quantify dynamic state transitions in cellular systems
* Quantified rates of muscle stem cell activation with single cell resolution for the first time using [*Heteromotility*](http://jck.bio/heteromotility)
* Developed [*Lanternfish*](http://jck.bio/lanternfish) deep learning package to enable discrimination of cell states from cell motility measurements and prediction of cell motility behaviors
* Demonstrated classification of stem cell differentiation states and cancerous transformation detection using [*Lanternfish*](http://jck.bio/lanternfish)
* Discovered [age-dampened state transitions in myogenesis](http://jck.bio/aging_musc_dynamics/) with single cell RNA-seq & [*Heteromotility*](http://jck.bio/heteromotility)

**IBM Research, Cell Engineering Group, San Jose, CA, 2017 Fall**  
*Deep Learning Research Intern*  
*Principal Investigator:* Simone Bianco  

* Developed rapid convolutional neural network (CNN) based image segmentation methods for processing of high-throughput timelapse imaging data
* Implemented a Natural Evolution Strategies (NES) optimization framework to improve multi-cell tracking
* Implemented a software interface to custom microscopy hardware, allowing for automation of high-throughput timelapse imaging experiments

<!-- **University of Central Florida, Orlando, FL, 2012 - 2015**  
*Burnett Research Scholar, Research and Mentoring Program Scholar*  
*Principal Investigator:* Kiminobu Sugaya, PhD

* Characterized induced pluripotent reprogramming in mesenchymal stem cells using machine-learning based image analysis, investigated cell sorting applications
* Developed a method to guide cell motility with polarized optical torques in collaboration with U. of Central Florida's optics research center
* Investigated a pathway for producing neuronal cells from adipose derived stem cells by modulation of Nanog

**University of California San Francisco, San Francisco, CA, 2014 Summer**  
*Amgen Scholar, Summer Research and Training Program*  
*Principal Investigator:* K. Mark Ansel, PhD

* Investigated the role of microRNA-29 in the regulation of cytokine production in T-cells

**National Institute on Aging, Baltimore, MD, 2012 Summer**  
*Summer Fellow, Image Informatics and Computational Biology Unit*
*Principal Investigator:* Ilya G. Goldberg, PhD

* Identified differential aging rates across the anatomy of *C. elegans* using machine learning based image classification
* Investigated the relationship between cognitive and age related morphologies in the primate hippocampus 

## Skills

**Data Analysis:** Experimental design, linear modeling, time series analysis, multivariate statistics  
**Machine Learning:** Semi-supervision, adversarial training, PyTorch, Tensorflow  
**NGS**: Single cell and bulk RNA-seq analysis, ATAC-seq, samtools, IGV  
**Programming:** Python (scipy, scikit-learn, statsmodels), R, Matlab, git, bash, LaTeX  
**Image Analysis:** Image segmentation, classification, particle tracking  
**Experimental Biology**: Quantitative microscopy, single cell RNA-seq, primary cell culture, FACS, molecular biology methods

-->

## Selected Publications

1. Roux A, Zhang C, Paw J, Zavala-Solorio J, Vijay T, Kolumam G, Kenyon C, **Kimmel JC**. 2022.
*Cell Systems*. [doi: https://doi.org/10.1016/j.cels.2022.05.002](https://doi.org/10.1016/j.cels.2022.05.002). [PDF]({{site.url}}/assets/files/2022_roux_cell_systems.pdf)
1. **Kimmel JC**, Kelley DR. scNym: Semi-supervised adversarial neural networks for single cell classification. 2021. *Genome Research*. [doi: https://doi.org/10.1101/gr.268581.120](https://doi.org/10.1101/gr.268581.120).
2. **Kimmel JC**, Yi N, Roy M, Hendrickson DG, Kelley DR. Differentiation reveals the plasticity of age-related change in murine muscle progenitors. 2021. *Cell Reports*. [https://doi.org/10.1016/j.celrep.2021.109046](https://doi.org/10.1016/j.celrep.2021.109046).
3. **Kimmel JC**, Hwang AB, Marshall WF, Brack AS. Aging induces aberrant state transition kinetics in murine muscle stem cells. 2020. *Development*. [https://doi.org/10.1242/dev.183855](doi: 10.1242/dev.183855). Chosen as a [Research Highlight by *Development*: Muscling in on Stem Cell Aging.](https://dev.biologists.org/content/147/9/e0901)
4. **Kimmel JC**. Disentangling latent representations of single cell RNA-seq experiments. 2020. *bioRxiv*. [https://doi.org/10.1101/2020.03.04.972166](doi: 10.1101/2020.03.04.972166).
5. **Kimmel JC**, Penland L, Rubinstein ND, Hendrickson DH, Kelley DR, Rosenthal AZ. A murine aging cell atlas reveals cell identity and tissue-specific trajectories of aging. 2019. *Genome Research*. [doi:
10.1101/gr.253880.119](https://genome.cshlp.org/content/29/12/2088). Featured on the cover of *Genome Research*.
6. **Kimmel JC**, Brack AS, Marshall WF. Deep convolutional and recurrent neural networks for cell motility discrimination and prediction. 2019. *IEEE Transactions on Computational Biology and Bioinformatics.* [doi: 10.1109/TCBB.2019.2919307.](https://pubmed.ncbi.nlm.nih.gov/31251191/) [Preprint](https://doi.org/10.1101/159202) featured in [*Company of Biologists: the Node*](http://thenode.biologists.com/july-in-preprints-2/highlights/).
7. **Kimmel JC**, Chang AY, Brack AS, Marshall WF. Inferring cell state by quantitative motility analysis reveals a dynamic state system and broken detailed balance. 2018. *PLoS Computational Biology* 14(1): e1005927. [https://doi.org/10.1371/journal.pcbi.1005927](https://doi.org/10.1371/journal.pcbi.1005927). Featured as an Editor's Pick in [*PLoS Editor's Collections: Cell Biology*](https://collections.plos.org/cell-biology-picks).
8. Constant C, **Kimmel JC**, Sugaya K, Dogariu A. Optically Controlled Subcellular Diffusion. 2015. *Frontiers in Optics & Laser Science*.

## Selected Presentations

1. Partial reprogramming restores youthful gene expression. **Invited speaker** at the Gordon Research Conference on *Systems Aging*. 2022.
2. Partial reprogramming restores youthful gene expression. **Invited speaker** at the *Longevity Summit*. Virtual. 2021.
<!-- 3. Partial reprogramming restores youthful gene expression. **Invited speaker** at the *Lifespan* conference. Virtual. 2021. -->
3. **Kimmel JC**, Kelley DR. scNym: Semi-supervised adversarial neural networks for single cell classification. **Selected speaker** at the International Conference on Machine Learning (ICML), Workshop on Computational Biology. Virtual. 2020. *Contributor Award for the best reviewed submissions*.
4. **Kimmel JC**, Kelley DR. scNym: Semi-supervised adversarial neural networks for single cell classification. **Selected speaker** at Intelligent Systems for Molecular Biology (ISMB), Machine Learning in Computational and Systems Biology session. Virtual. 2020.
5. **Kimmel JC**, Penland L, Rubinstein ND, Hendrickson DG, Kelley DR, Rosenthal AZ. Cell type and tissue-specific aging trajectories. **Invited speaker** for California QB3 Institute's *Aging and the Single Cell* event. San Francisco, CA. 2019.
6. **Kimmel JC**, Penland L, Rubinstein ND, Hendrickson DG, Kelley DR, Rosenthal AZ. Cell type and tissue-specific aging trajectories. **Invited speaker** at Mission Bay Capital Biolabs. San Francisco, CA. 2019.
7. **Kimmel JC**, Hwang A, Brack AS, Marshall WF. Inferring cell state dynamics with machine learning models. **Invited speaker** for the *Machine Learning in Cell Biology* Group meeting at ASCB-EMBO 2018. San Diego, CA. 2018.
<!-- 6. **Kimmel JC**, Chang AY, Brack AS, Marshall WF. Inferring stem cell states from cell motility behavior reveals a dynamic state system and broken detailed balance. **Selected speaker** at the Quantitative Biology meeting. Honolulu, HI. 2018. -->
8. **Kimmel JC**, Brack AS, Marshall WF. Deep neural networks for cell motility analysis. Poster presentation to Nvidia Deep Learning in Biomedicine Workshop. San Francisco, CA. 2018. *Nvidia Most Innovative Use of Deep Learning in Biomedicine Award.*


## Service

### Peer Reviewer

* *Bioinformatics*
* *Cell Reports*
* *Cell Systems*
* *eLife*
* *IEEE Journal of Biomedical and Health Informatics*
* *Nature Medicine*
* *Nature Methods*
* *PLoS Computational Biology*
* *Proceedings of the National Academy of Sciences*

### Open Source Software

* Maintainer: `scnym`, `velodyn`, `scmmd`, `heteromotility`, `lanternfish`, `pytorch_modelsize`
* Contributor: `scvi-tools`, `statsmodels`, `gseapy`

## Honors and Awards

* Calico Performance Recognition, 2020 & 2021
* ICML Workshop on Computational Biology, Contributor Award, 2020
* Selected as an *eLife* Early Career Reviewer, 2020
* Nvidia Best Presentation Award, *Nvidia Deep Learning in Biomedicine*, 2018
* National Science Foundation Graduate Research Fellowship, 2017
* PhRMA Informatics Fellowship, PhRMA Foundation, 2017
* Moritz-Heyman Discovery Fellow, UC San Francisco, 2017
* Nvidia GPU Grant, 2017 & 2018
* Amgen Scholar, University of California San Francisco, 2014
* Research & Mentoring Program Scholar, U. of Central Florida, 2013-2015
* Burnett Research Scholar, U. of Central Florida, 2013
* National Merit Scholar, National Merit Scholarship Corp., 2012
