---
title: CV
layout: page
permalink: /cv/index.html
---
## Jacob C. Kimmel
[jacob@jck.bio](mailto:jacob@jck.bio)

San Francisco, California

## Education

**Ph.D. â€” Dept. Biochemistry & Biophysics, UC San Francisco, 2015 - 2018**  
*Funding*: NSF Fellowship, PhRMA Fellowship, NIH T32, UCSF Discovery   

## Recent Experience

**NewLimit, South San Francisco, CA, 2022 - Present**  
*Co-founder & President*  

* Building a therapeutics firm to develop epigenetic reprogramming medicines for aging

**Calico Life Sciences, South San Francisco, CA, 2020 - 2022**  
*Principal Investigator, R&D, 2021-2022*  
*Computational Fellow, Computing, 2020-2021*

* Led a laboratory combining computational and experimental methods to repurpose developmental programs for therapeutic applications. 
* Developed [machine learning](https://scnym.research.calicolabs.com) and functional genomics methods to [measure and reprogram cell age & identity](https://reprog.research.calicolabs.com)

**Calico Life Sciences, South San Francisco, CA, 2018 - 2020**.  
*Data Scientist, Computing*

* Discovered that [cell identity rather than tissue environment defines the trajectory of cellular aging](https://mca.research.calicolabs.com) & a lineage decision where [old muscle stem cells fail to form mature muscle](https://myo.research.calicolabs.com)
* Contributed to small molecular and antibody therapeutics programs

**University of California San Francisco, San Francisco, CA, 2015 - 2018**  
*PhD Candidate*  
*Principal Investigators:* Wallace Marshall, Andrew Brack  
*Thesis:* Inferring stem cell state from cell behavior

* Developed [statistical](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005927) and [machine learning](https://ieeexplore.ieee.org/document/8747369) methods to quantify cellular behavior in timelapse microscopy images
* Discovered [state transitions in myogenesis that are impaired with age](http://jck.bio/aging_musc_dynamics/)

**IBM Research, Cell Engineering Group, San Jose, CA, 2017 Fall**  
*Deep Learning Research Intern*  
*Principal Investigator:* Simone Bianco  

* Developed convolutional neural networks to process timelapse imaging data for biological sensors
* Implemented a natural evolution strategies (NES) optimization framework to improve cellular tracking for biological sensors

## Selected Publications

1. Sengine L, Kummerlowe CS, Reynolds DL, Bernstein NJ, **Kimmel JC**. *In silico* design of epigenetic reprogramming payloads. *International Conference on Machine Learning (ICML)*, Generative Biology. 2025. url: [https://openreview.net/forum?id=kPQ6NKVAiT](https://openreview.net/forum?id=kPQ6NKVAiT).
2. Roux A, Zhang C, Paw J, Zavala-Solorio J, Vijay T, Kolumam G, Kenyon C, **Kimmel JC**. 2022.
Diverse partial reprogramming strategies restore youthful gene expression and transiently suppress cell identity.
*Cell Systems*. [doi: https://doi.org/10.1016/j.cels.2022.05.002](https://doi.org/10.1016/j.cels.2022.05.002). [PDF]({{site.url}}/assets/files/2022_roux_cell_systems.pdf)
3. **Kimmel JC**, Kelley DR. scNym: Semi-supervised adversarial neural networks for single cell classification. 2021. *Genome Research*. [doi: https://doi.org/10.1101/gr.268581.120](https://doi.org/10.1101/gr.268581.120). Awared Top Paper at ICML Computational Biology, 2020.
4. **Kimmel JC**, Yi N, Roy M, Hendrickson DG, Kelley DR. Differentiation reveals the plasticity of age-related change in murine muscle progenitors. 2021. *Cell Reports*. [https://doi.org/10.1016/j.celrep.2021.109046](https://doi.org/10.1016/j.celrep.2021.109046).
5. **Kimmel JC**, Hwang AB, Marshall WF, Brack AS. Aging induces aberrant state transition kinetics in murine muscle stem cells. 2020. *Development*. [https://doi.org/10.1242/dev.183855](doi: 10.1242/dev.183855). Chosen as a [Research Highlight by *Development*: Muscling in on Stem Cell Aging.](https://dev.biologists.org/content/147/9/e0901)
6. **Kimmel JC**. Disentangling latent representations of single cell RNA-seq experiments. 2020. *bioRxiv*. [https://doi.org/10.1101/2020.03.04.972166](doi: 10.1101/2020.03.04.972166).
7. **Kimmel JC**, Penland L, Rubinstein ND, Hendrickson DH, Kelley DR, Rosenthal AZ. A murine aging cell atlas reveals cell identity and tissue-specific trajectories of aging. 2019. *Genome Research*. [doi:
10.1101/gr.253880.119](https://genome.cshlp.org/content/29/12/2088). Featured on the cover of *Genome Research*.
8. **Kimmel JC**, Brack AS, Marshall WF. Deep convolutional and recurrent neural networks for cell motility discrimination and prediction. 2019. *IEEE Transactions on Computational Biology and Bioinformatics.* [doi: 10.1109/TCBB.2019.2919307.](https://pubmed.ncbi.nlm.nih.gov/31251191/) [Preprint](https://doi.org/10.1101/159202) featured in [*Company of Biologists: the Node*](http://thenode.biologists.com/july-in-preprints-2/highlights/).
9. **Kimmel JC**, Chang AY, Brack AS, Marshall WF. Inferring cell state by quantitative motility analysis reveals a dynamic state system and broken detailed balance. 2018. *PLoS Computational Biology* 14(1): e1005927. [https://doi.org/10.1371/journal.pcbi.1005927](https://doi.org/10.1371/journal.pcbi.1005927). Featured as an Editor's Pick in [*PLoS Editor's Collections: Cell Biology*](https://collections.plos.org/cell-biology-picks).

## Service

### Peer Reviewer

* *Bioinformatics*
* *Cell Reports*
* *Cell Systems*
* *eLife*
* *IEEE Journal of Biomedical and Health Informatics*
* *Nature Medicine*
* *Nature Methods*
* *PLoS Computational Biology*
* *Proceedings of the National Academy of Sciences*

### Open Source Software

* Maintainer: `scnym`, `velodyn`, `scmmd`, `heteromotility`, `lanternfish`, `pytorch_modelsize`
* Contributor: `scvi-tools`, `statsmodels`, `gseapy`